routers/repo/pull.go

```diff
@@ -582,6 +600,44 @@ func MergePullRequest(ctx *context.Context, form auth.MergePullRequestForm) {
 		return
 	}
 
+	// check if all required status checks are successful
```

Comment:
Since you're doing this check (the following 40 lines) the second time here (first time was in `routers/api/v1/repo/pull.go:581`), could you put that in a seperate function?

Reply:
done.